tests/gar.test.mjs

```diff
@@ -49,41 +51,366 @@ describe('GatewayRegistry', async () => {
 
     // assert no error tag
     const errorTag = transferResult.Messages?.[0]?.Tags?.find(
-      (tag) => tag.Name === 'Error',
+      (tag) => tag.name === 'Error',
     );
     assert.strictEqual(errorTag, undefined);
 
     return transferResult.Memory;
   };
 
-  beforeEach(async () => {
-    sharedMemory = await transfer({ sharedMemory });
+  const delegateStake = async ({
+    memory,
+    timestamp,
+    delegatorAddress,
+    quantity,
+    gatewayAddress,
+  }) => {
+    // give the wallet the delegate tokens
+    const transferMemory = await transfer({
+      recipient: delegatorAddress,
+      quantity,
+      memory,
+    });
+
+    const delegateResult = await handle(
+      {
+        From: delegatorAddress,
+        Owner: delegatorAddress,
+        Tags: [
+          { name: 'Action', value: 'Delegate-Stake' },
+          { name: 'Quantity', value: `${quantity}` }, // 2K IO
+          { name: 'Address', value: gatewayAddress }, // our gateway address
+        ],
+        Timestamp: timestamp,
+      },
+      transferMemory,
+    );
+
+    // assert no error tag
+    const errorTag = delegateResult.Messages?.[0]?.Tags?.find(
+      (tag) => tag.name === 'Error',
+    );
+    assert.strictEqual(errorTag, undefined);
+
+    return {
+      result: delegateResult,
+      memory: delegateResult.Memory,
+    };
```

Comment:
@arielmelendez - these functions wrap happy path handler behavior to set up the desired state easily within a test. See the updated tests for how they can be used to test various edge cases. The existing integration tests primarily focusing on validating happy path behavior of handlers, and inspecting various pieces of the resulting state (balances, gar, etc.). 

I'll look towards continuing to make integration test buildout easier across all test suites - this is just a starting point.